### ISSUE TITLE
Does `override` need to be overloaded explicitly?

### ISSUE THREAD

**Issue by bersbersbers** [mkorpela/overrides#109]
`mypy` treats `overrides` and `override` differently.

(Note that I reported `no-redef` separately as https://github.com/python/mypy/issues/13914. This issue is about the `misc` `mypy` message.)

```python
""""Bug."""
from typing import Callable, ParamSpec, TypeVar
P = ParamSpec("P")
T = TypeVar("T")

try:
    from overrides import override
except ModuleNotFoundError:
    def override(func: Callable[P, T]) -> Callable[P, T]:  # type: ignore[no-redef]
        return func

print(override)
```

> Incompatible redefinition (redefinition with type "Callable[[Callable[P, T]], Callable[P, T]]", original type overloaded function)  [misc]

By contrast, this is fine:

```python
""""No bug."""
from typing import Callable, ParamSpec, TypeVar
P = ParamSpec("P")
T = TypeVar("T")

try:
    from overrides import overrides
except ModuleNotFoundError:
    def overrides(func: Callable[P, T]) -> Callable[P, T]:  # type: ignore[no-redef]
        return func

print(overrides)
```

I wonder if this is a `mypy` issue or whether `override` needs to be overloaded explicitly.

**Comment by bersbersbers** [comment]
Not sure why, but
```python
F = TypeVar("F", bound=Callable)
override: Callable[[F], F]
```
solves both issues.

**Comment by mkorpela** [comment]
Thanks @bersbersbers !

**Comment by mkorpela** [comment]
Fixed in 3.7.1
